src/engine/GameObjects/MapNode.cxx

```diff
@@ -49,25 +49,53 @@ void MapNode::setBitmask(unsigned char elevationBitmask, unsigned char tileIDBit
 
 void MapNode::setTileID(const std::string &tileID)
 {
-  if (TileManager::instance().getTileData(tileID))
+  TileData *tileData = TileManager::instance().getTileData(tileID);
+  if (tileData)
   {
-    if (TileManager::instance().getTileData(tileID)->category == "Terrain" ||
-        TileManager::instance().getTileData(tileID)->category == "Water")
+    Layer layer = Layer::BUILDINGS;
+    if (tileData->category == "Terrain")
```

Comment:
String equality is pretty slow. Can we change this to an enum that gets serialized from JSON?

Reply:
At worst, we can use a `std::unordered_map` with string keys and lambdas for values. It would still be faster, but a worst solution than enum and switch-case